common/setups/returnn/datastreams/audio.py

```diff
@@ -0,0 +1,255 @@
+    "log_mel_filterbank": [type(None)],
+    "log_log_mel_filterbank": [type(None)],
+    "db_mel_filterbank": [type(None), DBMelFilterbankOptions],
+    "linear_spectrogram": [type(None), LinearFilterbankOptions],
+}
+
+
+class FeatureType(Enum):
+    """
+    Enum helper to have auto-completion for feature types
+    """
+
+    MFCC = "mfcc"
+    LOG_MEL_FILTERBANK = "log_mel_filterbank"
+    LOG_LOG_MEL_FILTERBANK = "log_log_mel_filterbank"
+    DB_MEL_FILTERBANK = "db_mel_filterbank"
+    LINEAR_SPECTROGRAM = "linear_spectrogram"
+
+
+@dataclasses.dataclass(frozen=True)
+class ReturnnAudioFeatureOptions:
+    """
+    Commonly used options for RETURNN feature extraction (e.g. with OggZip) and AudioFeatureDatastream
+
+    :param window_len:
+    :param step_len:
+    :param num_feature_filters:
+    :param with_delta:
+    :param features:
+    :param additional_feature_options:
+    :param sample_rate: audio sample rate, this is not strictly required for RETURNN itself
+        but might be needed for certain pipelines
+    :param peak_normalization:
+    :param preemphasis:
+    """
+
+    window_len: float = 0.025
+    step_len: float = 0.010
+    num_feature_filters: int = None
+    with_delta: bool = False
+    features: Union[str, FeatureType] = "mfcc"
+    feature_options: Optional[Union[dict, AdditionalFeatureOptions]] = None
+    sample_rate: Optional[int] = None
+    peak_normalization: bool = True
+    preemphasis: float = None
+
+    def __post_init__(self):
+        # convert Enum back to str
+        if isinstance(self.features, FeatureType):
+            # dataclass is frozen, so directly alter the self.__dict__
+            self.__dict__["features"] = self.features.value
+
+
+@dataclasses.dataclass(frozen=True)
+class ReturnnAudioRawOptions:
+    """
+    options for an AudioRawDatastream
+    """
+
+    peak_normalization: bool = True
+    preemphasis: float = None
+
+
+class AudioFeatureDatastream(Datastream):
+    """
+    Encapsulates options for audio features used by OggZipDataset via :class:`ExtractAudioFeaturesOptions` in RETURNN
+    """
+
+    def __init__(
+        self,
+        available_for_inference: bool,
+        options: ReturnnAudioFeatureOptions,
+        **kwargs,
+    ):
+        """
+        :param available_for_inference: define if the DataStream is available during decoding/search. If False,
+            it is only available during training.
+        :param options: An audio feature options object with the desired feature settings
+        :param kwargs: additional options that are passed manually
+        """
+        super().__init__(available_for_inference)
+        self.options = options
+        self.additional_options = kwargs.copy()
+
+        if options.features not in KNOWN_FEATURES:
+            print("Warning: %s is not a known feature type" % options.features)
+
+        if type(options.feature_options) not in KNOWN_FEATURES.get(options.features, [type(None)]):
```

Comment:
Since all types include `None` we can move the check down here and have `KNOWN_FEATURES` a bit cleaner.
If in the future we have features where this is not the case we can change this.